contrib/epee/include/string_tools.h

```diff
@@ -337,6 +338,11 @@ POP_WARNINGS
     s = *(t_pod_type*)bin_buff.data();
     return true;
   }
+	//----------------------------------------------------------------------------
+	inline bool validate_hex(uint64_t length, const std::string& str)
+	{
+		return std::regex_match(str, std::regex("'^[0-9a-fA-F]+$'")) && str.size() == length;
```

Comment:
We had some trouble in Kovri before about this kind of a-fA-F regex, as some locales don't sort the same. I'd spell out the whole set to avoid possible bugs there.